src/main/java/io/github/syst3ms/skriptparser/util/math/NumberMath.java

```diff
@@ -336,8 +373,41 @@ private static Number changeBoundExclusion(Number n, Number other) {
         }
     }
 
+    /**
+     * Java program to print all primes smaller than or equal to
+     * n using Sieve of Eratosthenes
+     */
+    private static List<BigInteger> sieveOfEratosthenes(int n) {
```

Comment:
If we've already generated the primes up to 100,000, and we want to compute the primes up to 100,100, we _really_ don't want to recompute all the primes below. As such, there should be an argument (which can be overloaded to 0) to specify from which value we want to start computing from. 
Then there'd be a new first pass that would immediately rule out all candidates divisible by the primes thus far. Note that you'd only need to do this first pass for primes below sqrt(n).